dev-games/ogre/ogre-1.10.12.ebuild

```diff
@@ -13,41 +15,46 @@ LICENSE="MIT public-domain"
 SLOT="0/1.10.0"
 KEYWORDS="~amd64 ~arm ~x86"
 
-IUSE="cg doc double-precision examples +freeimage gl3plus gles2 gles3 ois +opengl profile tools"
+IUSE="beta-components +cache cg doc double-precision egl examples +freeimage
+	gles2 ois +opengl profile +resman-legacy resman-pedantic resman-strict
+	tools"
 
-REQUIRED_USE="examples? ( ois )
-	gles3? ( gles2 )
-	gl3plus? ( opengl )"
+REQUIRED_USE="
+	|| ( gles2 opengl )
+	^^ ( resman-legacy resman-pedantic resman-strict )
```

Comment:
Apart from here, `resman-legacy` is nowhere used. You could drop this use flag, switch the condition to `??` instead and save one use flag without trouble.

Reply:
I thought a long time about this. Future code should support resman-strict, but current projects (and most of the examples) most probably need the resource manager to show its legacy behaviour.

So my first idea indeed was to use ??. But that makes it a bit opaque what is what. With three USE flags, with resman-legacy being the default, any user immediately can see from portage output, that there is an option that is not considered advisable per se.

However, as future code (or better: your projects resources) should be working fine with a pedantic or strict resource manager, developers must be given the opportunity to change the resource managers behaviour.

I know that it looks odd to have a default enabled use flag that is only used to show what you get with the other two disabled.
If I got rid of resman-legacy, I'd had to add its description to metadata.xml twice.

Reply:
I am currently switching to the ?? approach. Although I find it ugly to have to describe the legacy behaviour twice, devs who wish to upgrade their projects to support the newer resource manager behaviour should already know what they are doing.

Reply:
@Yamakuzure: The case-insensitivity being the focus here? It can be as simple as appending `(default: case-insensitive + sensitive lookup in all groups)`
Also a good opportunity to remove the plenks before colons.

Reply:
@a17r : That is a brillant idea and a lot better than what I currently have! Thanks you very much!